### PR TITLE
fix(server): correctly start idle timer

### PR DESCRIPTION
**Pull request by sentientwaffle**
Idle timer was only being triggered for client connections.

### CODE REVIEW

**Review by kincaidoneil — CHANGES_REQUESTED** [review]
I get that this is a potential DoS vector, but I actually like the current behavior, because it works better for longer-lived connections.

I wonder there's a way to get the best of both worlds?
- I don't like that connections would expire and then, if a client later sends a packet to it, it'd receive a Reject (thus causing their payment to fail & requiring them to query new credentials). Could we remove the expired connections behavior from `pool.ts`? So it still removes active connections when they close, but they'd just be recreated if the client sends more packets in the future.
- The only potential issue I can think of with that suggestion is the server could potentially reuse sequence numbers (bad). In the send-only case, this isn't an issue, since the server can never send packets to the client. But if you think this is an issue, we could do something on the initial Prepare where we set `nextPacketSequence` to the incoming Prepare's sequence number + 1 or something -- because it's up to the sender not to reuse sequence numbers.
- When the connection expires, this will call `destroy()` and trigger the server to send a `ConnectionClose` packet back to the client, which with the way Pay works by sharing a client address of `''`, I think this will throw but get caught, so it's probably okay.

**Comment by sentientwaffle** [comment]
Wouldn't it be simpler to just use a higher idle timeout on the server side?

**Comment by kincaidoneil** [comment]
I mean, I'd like connections to be able to stay "open" for days on end... but I also don't think we should decide that now by way of a default parameter.

And on second thought, I don't think the sequence number thing is an issue: if it's a bidirectional connection, the client already shared their address and probably receives the connection close, thus closing their connection. If it's a send-only client, they won't get any connection close, but if they send more packets later the server still won't know their ILP address and can never send them outgoing packets, so there's no risk of sequence number reuse.

The removal of AgingSet/expired connections should be pretty minor?

File: src/pool.ts [hunk 1]
```diff
@@ -3,7 +3,6 @@ import * as IlpPacket from 'ilp-packet'
 import { Reader } from 'oer-utils'
 import { Connection, BuildConnectionOpts } from './connection'
 import * as cryptoHelper from './crypto'
-import { AgingSet } from './util/aging-set'
 
 const log = createLogger('ilp-protocol-stream:Pool')
 const CLOSED_CONNECTION_CYCLE = 10 * 60 * 1000 // 10 minutes
```
Comment (kincaidoneil) [inline src/pool.ts:8]:
```suggestion
```

File: test/index.test.ts [hunk 2]
```diff
@@ -307,7 +307,7 @@ describe('Server', function () {
     it('should reject packets for connections that have already been closed', async function () {
```
Comment (kincaidoneil) [inline test/index.test.ts:307]:
```suggestion
    it('should create a new connection if client sends more packets after connection is closed', async function () {
```

**Review by kincaidoneil — APPROVED** [review]

**Pull request merged**